### DISCUSSION TITLE
black code formatter

### DISCUSSION THREAD

**Discussion by NimaSarajpoor** [TDAmeritrade/stumpy#594] [Help: Other Questions] [answered]
I have been trying to run `./test.sh` which runs the `black` code formatter as well. However, I faced the following error:
```
(base) nimas (main) stumpy
$ ./test.sh
Cleaning Up
Checking Black Code Formatting
Traceback (most recent call last):
  File "C:\Users\nimas\anaconda3\Scripts\black-script.py", line 6, in <module>
    from black import patched_main
  File "src\black\__init__.py", line 41, in <module>
  File "src\black\nodes.py", line 36, in <module>
  File "src\blib2to3\pygram.py", line 186, in initialize
  File "src\blib2to3\pygram.py", line 32, in __init__
AttributeError: '_python_symbols' object has no attribute 'with_item'
Error: pytest encountered exit code 1
```
Updating `conda` and `black` did not help. Note sure why I get this error. (my version of black is 22)

Installing packages of `enviornment.yml` in a new environment resolved the issue...
```
conda env update --name myenv --file environment.yml --prune
```

Not sure if it would be a good  idea to add the command above to the contribution notebook.

**Comment by seanlaw** [comment, marked as answer]
I have never seen that before but, based on your brief description, it sounds like a corrupted environment. I start with a fresh environment frequently and treat them as fungible. I would **not** add this to the Contributors guide unless it was a common problem.

**Reply by NimaSarajpoor** [reply]
> I would not add this to the Contributors guide unless it was a common problem.

True! That is why I was hesitant about this and opened a discussion rather than issue. 

>  I start with a fresh environment frequently and treat them as fungible.
Thanks for the advice!
